workflow_scripts/update_onnx_hub_manifest.py

```diff
@@ -0,0 +1,122 @@
+import glob
+import hashlib
+import json
+import os
+import re
+import bs4
+import markdown
+import pandas as pd
+import typepy
+from os.path import join, split
+
+
+# Acknowledgments to pytablereader codebase for this function
+def parse_html(table):
+    headers = []
+    data_matrix = []
+    rows = table.find_all("tr")
+    re_table_val = re.compile("td|th")
+    for row in rows:
+        td_list = row.find_all("td")
+        if typepy.is_empty_sequence(td_list):
+            if typepy.is_not_empty_sequence(headers):
+                continue
+            th_list = row.find_all("th")
+            if typepy.is_empty_sequence(th_list):
+                continue
+            headers = [row.text.strip() for row in th_list]
+            continue
+        data_matrix.append(list(row.find_all(re_table_val)))
+
+    if typepy.is_empty_sequence(data_matrix):
+        raise ValueError("data matrix is empty")
+
+    return pd.DataFrame(data_matrix, columns=headers)
+
+
+markdown_files = list(glob.glob(join("..", "**", "README.md"), recursive=True))[1:]
+manifest_object = []
+
+all_tables = []
+for markdown_file in markdown_files:
+    with open(markdown_file, "r") as f:
+        parsed = markdown.markdown(f.read(), extensions=["markdown.extensions.tables"])
+        soup = bs4.BeautifulSoup(parsed, "html.parser")
+        for table in soup.find_all("table"):
+            parsed = parse_html(table)
+            if all(col in parsed.columns.values for col in ["Model", "Download", "Opset version", "ONNX version"]):
+                parsed["source_file"] = markdown_file
+                all_tables.append(parsed)
+            else:
+                print("Unrecognized table columns in file {}: {}".format(markdown_file, parsed.columns.values))
+
+df = pd.concat(all_tables, axis=0)
+normalize_name = {
+    'Download': "model_path",
+    'Download (with sample test data)': 'model_with_data_path',
+}
+
+top_level_fields = [
+    "model", "model_path", "opset_version", "onnx_version"
+]
+
+
+def prep_name(col):
+    if col in normalize_name:
+        col = normalize_name[col]
+    col = col.rstrip()
+    prepped_col = col.replace(" ", "_").lower()
+    if prepped_col in top_level_fields:
+        return prepped_col
+    else:
+        return col
+
+
+renamed = df.rename(columns={col: prep_name(col) for col in df.columns.values})
+metadata_fields = [f for f in renamed.columns.values if f not in top_level_fields]
+
+
+def get_file_info(row, feild):
+    source_dir = split(row["source_file"])[0]
+    model_file = row[feild].contents[0].attrs['href']
+    rel_path = os.sep.join(join(source_dir, model_file).split(os.sep)[1:])
+    with open(join("..", rel_path), "rb") as f:
+        bytes = f.read()
+        sha256 = hashlib.sha256(bytes).hexdigest()
+    return {
+        feild: rel_path,
+        feild.replace("_path", "") + "_sha": sha256,
+        feild.replace("_path", "") + "_bytes": len(bytes)}
+

```

Comment:
Add a function to generate tags based on folder structure.
```suggestion
def get_model_tags(row):
    source_dir = split(row["source_file"])[0]
    raw_tags = source_dir.split(os.sep)[1:]
    return [tag.replace("_", " ") for tag in raw_tags]
```